cedar-lean/Cedar/Spec/Policy.lean

```diff
@@ -99,14 +110,23 @@ def ActionScope.toExpr : ActionScope → Expr
     let exprs := es.map (fun e => .lit (.entityUID e))
     .binaryApp (.mem) (.var .action) (.set exprs)
 
+def Condition.toExpr (c : Condition) : Expr :=
+  match c.kind with
+  | .when => c.body
+  | .unless => Expr.unaryApp .not c.body
+
+-- Conditions are evaluated top to bottom, and short circuit
+def conditionsToExpr (cs : Conditions) : Expr :=
```

Comment:
We can write this using typeclass inference to avoid some wrapping.  Writing `Conditions.toExpr` means we can also write `cs.toExpr` in the client code when Lean knows that `cs` is of type `Conditions` (see below):

```suggestion
def Conditions.toExpr (cs : Conditions) : Expr :=
  cs.foldr (λ c expr => .and c.toExpr expr) true
```